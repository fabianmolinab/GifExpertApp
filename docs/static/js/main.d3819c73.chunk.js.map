{"version":3,"sources":["components/AddCategory.jsx","helpers/getGifs.js","components/GifGidItem.jsx","components/GifGrid.js","hooks/useFetchGifs.js","GifExpertApp.jsx","index.js"],"names":["AddCategory","setCategories","useState","inputValue","setInputValue","onSubmit","e","preventDefault","trim","length","cats","type","value","onChange","target","getGifs","category","a","url","encodeURI","fetch","resp","json","data","gifs","map","img","id","title","images","downsized_medium","GifGidItem","className","src","alt","GifGrid","loading","state","setState","useEffect","then","imgs","catch","error","console","log","useFetchGifs","GifExpertApp","categorias","ReactDOM","render","document","getElementById"],"mappings":"mMAGaA,EAAc,SAAC,GAAqB,IAApBC,EAAmB,EAAnBA,cAAmB,EAEVC,mBAAS,IAFC,mBAEvCC,EAFuC,KAE3BC,EAF2B,KAqB9C,OACE,sBAAMC,SAZa,SAACC,GACpBA,EAAEC,iBAGGJ,EAAWK,OAAOC,QAAU,IAE/BR,GAAc,SAAAS,GAAI,OAAKP,GAAL,mBAAoBO,OACtCN,EAAc,MAKhB,SACE,uBACEO,KAAK,OACLC,MAAOT,EACPU,SArBoB,SAACP,GACzBF,EAAcE,EAAEQ,OAAOF,a,+BCPdG,EAAO,uCAAG,WAAOC,GAAP,uBAAAC,EAAA,6DAEfC,EAFe,iDAEiCC,UAAWH,GAF5C,+DAGFI,MAAMF,GAHJ,cAGfG,EAHe,gBAIEA,EAAKC,OAJP,uBAIbC,EAJa,EAIbA,KAEFC,EAAOD,EAAKE,KAAI,SAAAC,GAAQ,IAAD,EAC3B,MAAO,CACLC,GAAID,EAAIC,GACRC,MAAOF,EAAIE,MACXV,IAAG,UAAEQ,EAAIG,cAAN,aAAE,EAAYC,iBAAiBZ,QAVjB,kBAcdM,GAdc,4CAAH,sDCAPO,EAAa,SAAC,GAAiB,IAAhBH,EAAe,EAAfA,MAAMV,EAAS,EAATA,IAChC,OACE,qBAAKc,UAAU,yCAAf,SAEE,qBACEC,IAAKf,EACLgB,IAAKN,OCHAO,EAAU,SAAC,GAAkB,IAAhBnB,EAAe,EAAfA,SAAe,ECFb,SAACA,GAAc,IAAD,EAEdd,mBAAS,CACjCqB,KAAM,GACNa,SAAS,IAJ6B,mBAEjCC,EAFiC,KAE1BC,EAF0B,KAkBxC,OAXAC,qBAAW,WACTxB,EAAQC,GACLwB,MAAM,SAAAC,GACLH,EAAS,CACPf,KAAMkB,EACNL,SAAS,OAGZM,OAAO,SAAAC,GAAK,OAAIC,QAAQC,IAAIF,QAC9B,CAAE3B,IAEEqB,EDbwBS,CAAc9B,GAAjCa,EAH2B,EAGhCN,KAAaa,EAHmB,EAGnBA,QAEpB,OACE,qCACE,mCAAMpB,EAAN,OAEEoB,GAAW,yCAEb,qBAAKJ,UAAU,YAAf,SAEIH,EAAOJ,KAAK,SAAAC,GAAG,OACb,cAAC,EAAD,eAEOA,GADEA,YEhBRqB,EAAe,WAAO,IAAD,EAEI7C,mBAAS,IAFb,mBAEzB8C,EAFyB,KAEb/C,EAFa,KAIhC,OACE,qCACE,8CACA,cAAC,EAAD,CAAaA,cAAgBA,IAC7B,uBAEA,6BAEI+C,EAAWvB,KAAK,SAAAT,GAAQ,OACtB,cAAC,EAAD,CACEA,SAAWA,GACLA,Y,MCZpBiC,IAASC,OACP,cAAC,EAAD,IACAC,SAASC,eAAe,W","file":"static/js/main.d3819c73.chunk.js","sourcesContent":["import React, { useState } from 'react'\nimport PropTypes from 'prop-types';\n\nexport const AddCategory = ({setCategories}) => {\n\n  const [inputValue, setInputValue] = useState('');\n\n// Cada vez que escriba\n  const handleInputChange = (e) => {\n    setInputValue(e.target.value)\n  }\n\n// Cuando haces ENTER\n  const handleSubmit = (e) => {\n    e.preventDefault(); // Previene el refrescar de la pagina  \n    \n    // ValidaciÃ³n del campo de busqueda\n    if ( inputValue.trim().length >= 2 ) {\n\n      setCategories(cats => [inputValue, ...cats]);\n      setInputValue('');\n    }\n    \n  }\n  return (\n    <form onSubmit={handleSubmit}>\n      <input\n        type=\"text\"\n        value={inputValue}\n        onChange={handleInputChange}\n      />\n    </form>\n  )\n};\nAddCategory.propTypes = {\n\n  setCategories: PropTypes.func.isRequired\n\n}","\n\nexport const getGifs = async (category) => {\n\n  const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI( category )}&limit=10&api_key=mRT7lkeqFhhlrf6GTm6lBi2yLCmn8o66`\n  const resp = await fetch(url);\n  const { data } = await resp.json();\n\n  const gifs = data.map(img => {\n    return {\n      id: img.id,\n      title: img.title,\n      url: img.images?.downsized_medium.url\n    }\n  })\n\n  return gifs;\n}","import React from 'react'\n\nexport const GifGidItem = ({title,url}) => {\n  return (\n    <div className=\"card animate__animated animate__wobble\">\n      \n      <img \n        src={url} \n        alt={title}\n      />\n    </div>\n  )\n}\n\n","import React from 'react';\nimport { useFetchGifs } from '../hooks/useFetchGifs';\nimport { GifGidItem } from './GifGidItem';\n\n\nexport const GifGrid = ({ category }) => {\n\n  \n  const {data:images, loading} = useFetchGifs( category ); // Llamado a la API  \n\n  return (\n    <>\n      <h3> {category} </h3>\n\n      { loading && <p>Cargando</p> } \n\n      <div className=\"card-grid\">  \n        {\n          images.map( img => (\n            <GifGidItem \n              key= { img } \n              { ...img }\n            />\n          ))\n        }\n      </div>\n    </>\n  )\n};","import { useState, useEffect } from 'react'; \nimport {getGifs} from '../helpers/getGifs';\n\nexport const useFetchGifs = (category) => {\n\n  const [state, setState] = useState({  \n    data: [],\n    loading: true\n  });\n\n  useEffect( () => {\n    getGifs(category) //Peticion a la API\n      .then( imgs =>{\n        setState({ \n          data: imgs,\n          loading: false\n        });          \n      }) \n      .catch( error => console.log(error))\n  }, [ category ]);\n  \n  return state; //{data: [], loading: true};\n}","import React, { useState } from 'react';\nimport { AddCategory } from './components/AddCategory';\nimport { GifGrid } from './components/GifGrid';\n\nexport const GifExpertApp = () => {\n  \n  const [categorias, setCategories] = useState([]);\n\n  return ( \n    <>\n      <h2>GifExpertApp</h2> \n      <AddCategory setCategories={ setCategories }/>\n      <hr/>\n\n      <ol>\n        {\n          categorias.map( category => \n            <GifGrid \n              category={ category }\n              key={ category }\n            />\n          )\n        }\n      </ol>\n    </>\n  )\n};","import React from 'react';\nimport ReactDOM from 'react-dom';\n\nimport { GifExpertApp } from './GifExpertApp';\n\nimport './index.css';\n\nReactDOM.render(\n  <GifExpertApp />,\n  document.getElementById('root')\n);"],"sourceRoot":""}