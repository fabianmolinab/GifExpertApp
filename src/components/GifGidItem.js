import React from 'react';
import PropTypes from 'prop-types';

export const GifGidItem = ({title,url}) => {
  return (
    <div className="card animate__animated animate__wobble">
      
      <img 
        src={url} 
        alt={title}
      />
    </div>
  )
};

GifGidItem.propTypes = {

  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired

}

