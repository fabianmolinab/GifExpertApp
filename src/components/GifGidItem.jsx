import React from 'react'

export const GifGidItem = ({title,url}) => {
  return (
    <div className="card animate__animated animate__wobble">
      
      <img 
        src={url} 
        alt={title}
      />
    </div>
  )
}

