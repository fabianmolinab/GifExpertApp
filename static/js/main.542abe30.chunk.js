(this["webpackJsonpgif-expert-app"]=this["webpackJsonpgif-expert-app"]||[]).push([[0],{17:function(e,t,n){},18:function(e,t,n){"use strict";n.r(t);var r=n(0),c=n(1),a=n(7),i=n.n(a),s=n(2),u=n(9),o=function(e){var t=e.setCategories,n=Object(c.useState)(""),a=Object(s.a)(n,2),i=a[0],o=a[1];return Object(r.jsxs)("form",{onSubmit:function(e){e.preventDefault(),i.trim().length>=2&&(t((function(e){return[i].concat(Object(u.a)(e))})),o(""))},children:[Object(r.jsxs)("p",{children:[" ",i," "]}),Object(r.jsx)("input",{type:"text",value:i,onChange:function(e){o(e.target.value)}})]})},j=n(10),l=n(5),d=n.n(l),b=n(8),f=function(){var e=Object(b.a)(d.a.mark((function e(t){var n,r,c,a,i;return d.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return n="https://api.giphy.com/v1/gifs/search?q=".concat(encodeURI(t),"&limit=50&api_key=mRT7lkeqFhhlrf6GTm6lBi2yLCmn8o66"),e.next=3,fetch(n);case 3:return r=e.sent,e.next=6,r.json();case 6:return c=e.sent,a=c.data,i=a.map((function(e){var t;return{id:e.id,title:e.title,url:null===(t=e.images)||void 0===t?void 0:t.downsized_medium.url}})),e.abrupt("return",i);case 10:case"end":return e.stop()}}),e)})));return function(t){return e.apply(this,arguments)}}(),p=function(e){var t=e.title,n=e.url;return Object(r.jsx)("div",{className:"card animate__animated animate__wobble",children:Object(r.jsx)("img",{src:n,alt:t})})},h=function(e){var t=e.category,n=function(e){var t=Object(c.useState)({data:[],loading:!0}),n=Object(s.a)(t,2),r=n[0],a=n[1];return Object(c.useEffect)((function(){f(e).then((function(e){a({data:e,loading:!1})})).catch((function(e){return console.log(e)}))}),[e]),r}(t),a=n.data,i=n.loading;return Object(r.jsxs)(r.Fragment,{children:[Object(r.jsxs)("h3",{children:[" ",t," "]}),i&&Object(r.jsx)("p",{children:"Cargando"}),Object(r.jsx)("div",{className:"card-grid",children:a.map((function(e){return Object(r.jsx)(p,Object(j.a)({},e),e)}))})]})},O=function(e){var t=e.defaulCategories,n=void 0===t?[]:t,a=Object(c.useState)(n),i=Object(s.a)(a,2),u=i[0],j=i[1];return Object(r.jsxs)(r.Fragment,{children:[Object(r.jsx)("h2",{children:"\xa1Puedes buscar cualquier Gif!"}),Object(r.jsx)(o,{setCategories:j}),Object(r.jsx)("hr",{}),Object(r.jsx)("ol",{children:u.map((function(e){return Object(r.jsx)(h,{category:e},e)}))})]})};n(17);i.a.render(Object(r.jsx)(O,{}),document.getElementById("root"))}},[[18,1,2]]]);
//# sourceMappingURL=main.542abe30.chunk.js.map