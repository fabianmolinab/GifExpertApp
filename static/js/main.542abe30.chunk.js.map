{"version":3,"sources":["components/AddCategory.js","helpers/getGifs.js","components/GifGidItem.js","components/GifGrid.js","hooks/useFetchGifs.js","GifExpertApp.js","index.js"],"names":["AddCategory","setCategories","useState","inputValue","setInputValue","onSubmit","e","preventDefault","trim","length","cats","type","value","onChange","target","getGifs","category","a","url","encodeURI","fetch","resp","json","data","gifs","map","img","id","title","images","downsized_medium","GifGidItem","className","src","alt","GifGrid","loading","state","setState","useEffect","then","imgs","catch","error","console","log","useFetchGifs","GifExpertApp","defaulCategories","categorias","ReactDOM","render","document","getElementById"],"mappings":"mMAGaA,EAAc,SAAC,GAAuB,IAArBC,EAAoB,EAApBA,cAAoB,EAEZC,mBAAS,IAFG,mBAEzCC,EAFyC,KAE7BC,EAF6B,KAoBhD,OACE,uBAAMC,SAXa,SAACC,GACpBA,EAAEC,iBAGEJ,EAAWK,OAAOC,QAAU,IAE9BR,GAAc,SAAAS,GAAI,OAAKP,GAAL,mBAAoBO,OACtCN,EAAc,MAIhB,UACE,kCAAKD,EAAL,OACA,uBACEQ,KAAK,OACLC,MAAOT,EACPU,SArBoB,SAACP,GACzBF,EAAcE,EAAEQ,OAAOF,c,+BCPdG,EAAO,uCAAG,WAAOC,GAAP,uBAAAC,EAAA,6DAEfC,EAFe,iDAEiCC,UAAWH,GAF5C,+DAGFI,MAAMF,GAHJ,cAGfG,EAHe,gBAIEA,EAAKC,OAJP,uBAIbC,EAJa,EAIbA,KAEFC,EAAOD,EAAKE,KAAI,SAAAC,GAAQ,IAAD,EAC3B,MAAO,CACLC,GAAID,EAAIC,GACRC,MAAOF,EAAIE,MACXV,IAAG,UAAEQ,EAAIG,cAAN,aAAE,EAAYC,iBAAiBZ,QAVjB,kBAcdM,GAdc,4CAAH,sDCCPO,EAAa,SAAC,GAAiB,IAAhBH,EAAe,EAAfA,MAAMV,EAAS,EAATA,IAChC,OACE,qBAAKc,UAAU,yCAAf,SAEE,qBACEC,IAAKf,EACLgB,IAAKN,OCFAO,EAAU,SAAC,GAAkB,IAAhBnB,EAAe,EAAfA,SAAe,ECJb,SAACA,GAAc,IAAD,EAEdd,mBAAS,CACjCqB,KAAM,GACNa,SAAS,IAJ6B,mBAEjCC,EAFiC,KAE1BC,EAF0B,KAkBxC,OAXAC,qBAAW,WACTxB,EAAQC,GACLwB,MAAM,SAAAC,GACLH,EAAS,CACPf,KAAMkB,EACNL,SAAS,OAGZM,OAAO,SAAAC,GAAK,OAAIC,QAAQC,IAAIF,QAC9B,CAAE3B,IAEEqB,EDX0BS,CAAa9B,GAAjCa,EAH0B,EAG/BN,KAAaa,EAHkB,EAGlBA,QAErB,OACE,qCACE,mCAAMpB,EAAN,OAEEoB,GAAW,yCAEb,qBAAKJ,UAAU,YAAf,SAEIH,EAAOJ,KAAI,SAAAC,GAAG,OACZ,cAAC,EAAD,eAEMA,GADCA,YElBNqB,EAAe,SAAC,GAA+B,IAAD,IAA5BC,wBAA4B,MAAT,GAAS,IAErB9C,mBAAS8C,GAFY,mBAElDC,EAFkD,KAEtChD,EAFsC,KAIzD,OACE,qCACE,kEACA,cAAC,EAAD,CAAaA,cAAeA,IAC5B,uBAEA,6BAEIgD,EAAWxB,KAAI,SAAAT,GAAQ,OACrB,cAAC,EAAD,CACEA,SAAUA,GACLA,Y,MCZnBkC,IAASC,OACP,cAAC,EAAD,IACAC,SAASC,eAAe,W","file":"static/js/main.542abe30.chunk.js","sourcesContent":["import React, { useState } from 'react'\nimport PropTypes from 'prop-types';\n\nexport const AddCategory = ({ setCategories }) => {\n\n  const [inputValue, setInputValue] = useState('');\n\n  // Cada vez que escriba\n  const handleInputChange = (e) => {\n    setInputValue(e.target.value)\n  }\n\n  // Cuando haces ENTER\n  const handleSubmit = (e) => {\n    e.preventDefault(); // Previene el refrescar de la pagina\n\n    // Validación del campo de busqueda\n    if (inputValue.trim().length >= 2) {\n\n      setCategories(cats => [inputValue, ...cats]);\n      setInputValue('');\n    }\n  }\n  return (\n    <form onSubmit={handleSubmit}>\n      <p> {inputValue} </p>\n      <input\n        type=\"text\"\n        value={inputValue}\n        onChange={handleInputChange}\n      />\n    </form>\n  )\n};\nAddCategory.propTypes = {\n\n  setCategories: PropTypes.func.isRequired\n\n};","\n\nexport const getGifs = async (category) => {\n\n  const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI( category )}&limit=50&api_key=mRT7lkeqFhhlrf6GTm6lBi2yLCmn8o66`\n  const resp = await fetch(url);\n  const { data } = await resp.json();\n\n  const gifs = data.map(img => {\n    return {\n      id: img.id,\n      title: img.title,\n      url: img.images?.downsized_medium.url\n    }\n  })\n\n  return gifs;\n}","import React from 'react';\nimport PropTypes from 'prop-types';\n\nexport const GifGidItem = ({title,url}) => {\n  return (\n    <div className=\"card animate__animated animate__wobble\">\n      \n      <img \n        src={url} \n        alt={title}\n      />\n    </div>\n  )\n};\n\nGifGidItem.propTypes = {\n\n  title: PropTypes.string.isRequired,\n  url: PropTypes.string.isRequired\n\n}\n\n","import React from 'react';\nimport PropTypes from 'prop-types'\n\nimport { useFetchGifs } from '../hooks/useFetchGifs';\nimport { GifGidItem } from './GifGidItem';\n\n\nexport const GifGrid = ({ category }) => {\n\n\n  const { data:images, loading } = useFetchGifs(category); // Llamado a la API  \n\n  return (\n    <>\n      <h3> {category} </h3>\n\n      { loading && <p>Cargando</p>}\n\n      <div className=\"card-grid\">\n        {\n          images.map(img => (\n            <GifGidItem\n              key={img}\n              {...img}\n            />\n          ))\n        }\n      </div>\n    </>\n  )\n}\n\nGifGrid.propTypes = {\n\n  category: PropTypes.string.isRequired\n\n}\n","import { useState, useEffect } from 'react'; \nimport {getGifs} from '../helpers/getGifs';\n\nexport const useFetchGifs = (category) => {\n\n  const [state, setState] = useState({  \n    data: [],\n    loading: true\n  });\n\n  useEffect( () => {\n    getGifs(category) //Peticion a la API\n      .then( imgs =>{\n        setState({ \n          data: imgs,\n          loading: false\n        });          \n      }) \n      .catch( error => console.log(error))\n  }, [ category ]);\n  \n  return state; //{data: [], loading: true};\n}","import React, { useState } from 'react';\nimport { AddCategory } from './components/AddCategory';\nimport { GifGrid } from './components/GifGrid';\n\nexport const GifExpertApp = ({ defaulCategories = [] }) => {\n\n  const [categorias, setCategories] = useState(defaulCategories);\n\n  return (\n    <>\n      <h2>¡Puedes buscar cualquier Gif!</h2>\n      <AddCategory setCategories={setCategories} />\n      <hr />\n\n      <ol>\n        {\n          categorias.map(category =>\n            <GifGrid\n              category={category}\n              key={category}\n            />\n          )\n        }\n      </ol>\n    </>\n  )\n};","import React from 'react';\nimport ReactDOM from 'react-dom';\n\nimport { GifExpertApp } from './GifExpertApp';\n\nimport './index.css';\n\nReactDOM.render(\n  <GifExpertApp />,\n  document.getElementById('root')\n);"],"sourceRoot":""}