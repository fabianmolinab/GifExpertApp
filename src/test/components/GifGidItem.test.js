import React from 'react';

import {shallow} from 'enzyme';
import { GifGidItem } from '../../components/GifGidItem.js';

describe( 'Pruebas en <GifgridItem/>', () => {

  const title = "Fabian";
  const url   = "https://www.local.com";
  const wrapper = shallow ( <GifGidItem 
    title={ title } 
    url= { url }
    />);

  test( 'Debe mostrar el componente correctamente', () => {    
    
    expect( wrapper ).toMatchSnapshot();

  });
  test('Debe tener una img con la url y alt de los props', () => {

    const img = wrapper.find('img');
    
    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);
  });

  test('El div debe tener el className animate__wobble', () => {
    const div = wrapper.find('div');

    expect(div.props().className.includes("animate__wobble")).toBe(true);

  })
});