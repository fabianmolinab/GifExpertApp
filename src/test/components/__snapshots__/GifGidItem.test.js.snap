// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifgridItem/> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__wobble"
>
  <img
    alt="Fabian"
    src="https://www.local.com"
  />
</div>
`;
